core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1731,6 +1764,57 @@ object Parser {
     */
   def end: Parser0[Unit] = Impl.EndParser
 
+  /** return the input string we are parsing.
+    */
+  def input: Parser0[String] = Impl.Input
+
+  /** return the position the parser is at
+    */
+  def careting: Parser0[Caret] =
+    (input, index).tupled map { case (in, ix) =>
```

Comment:
please avoid infix use of methods (deprecated in scala 3).

Can we instead do `(input, index).mapN { (in, ix) => ...`

Reply:
Oh sorry, I only read the Scala 2 style guide and saw that the infix form *was* recommended...